src/routers/StackRouter.js

```diff
@@ -139,7 +139,7 @@ export default (
       }
 
       // Check if a child scene wants to handle the action as long as it is not a reset to the root stack
-      if (action.type !== NavigationActions.RESET || action.key !== null) {
+      if (action.type !== NavigationActions.RESET || !!action.key) {
```

Comment:
I think it would be best to be explicit here and not use `!!` because this will coerce values like `0` into `false` where `0` might be a valid key in some instances (either in the current version or in future versions).